keras_rs/src/layers/embedding/jax/distributed_embedding.py

```diff
@@ -0,0 +1,1033 @@
+            )
+            return out
+
+    def set_embedding_tables(self, tables: Mapping[str, ArrayLike]) -> None:
+        """Sets the embedding tables to specific (unsharded) values.
+
+        Args:
+          tables: Mapping of table name -> table values.
+        """
+        if "default_device" in self._placement_to_path_to_feature_config:
+            self._default_device_set_tables(tables)
+
+        if "sparsecore" in self._placement_to_path_to_feature_config:
+            self._sparsecore_set_tables(tables)
+
+    def _default_device_set_tables(
+        self, tables: Mapping[str, ArrayLike]
+    ) -> None:
+        if not self.built:
+            raise ValueError("Layer must first be built before setting tables.")
+
+        if "default_device" in self._placement_to_path_to_feature_config:
+            table_to_embedding_layer = {}
+            for (
+                path,
+                feature_config,
+            ) in self._placement_to_path_to_feature_config[
+                "default_device"
+            ].items():
+                table_to_embedding_layer[feature_config.table] = (
+                    self._default_device_embedding_layers[path]
+                )
+
+            for table, embedding_layer in table_to_embedding_layer.items():
+                table_values = tables.get(table.name, None)
+                if table_values is not None:
+                    if embedding_layer.lora_enabled:
+                        raise ValueError("Cannot set table if LoRA is enabled.")
+                    # pylint: disable-next=protected-access
+                    embedding_layer._embeddings.assign(table_values)
+
+    def _sparsecore_set_tables(self, tables: Mapping[str, ArrayLike]) -> None:
+        if not self._sparsecore_built:
+            self._sparsecore_build()
+
+        config = self._config
+        num_table_shards = config.mesh.devices.size * config.num_sc_per_device
+        table_specs = embedding_utils.get_table_specs(config.feature_specs)
+        sharded_tables = embedding_utils.stack_and_shard_tables(
+            table_specs,
+            tables,
+            num_table_shards,
+        )
+
+        device_tables = jax.device_put(
+            jax.tree.map(
+                # Flatten shard dimension to allow auto-sharding to split the
+                # array.
+                lambda table: table.reshape((-1, table.shape[-1])),
+                sharded_tables,
+            ),
+            self._sparsecore_layout.backend_layout,
+        )
+
+        # Assign stacked table variables to the device values.
+        keras.tree.map_structure_up_to(
+            device_tables,
+            lambda table_and_slot_variables,
+            table_value: table_and_slot_variables[0].assign(table_value),
+            self._table_and_slot_variables,
+            device_tables,
+        )
+
+    def get_embedding_tables(self) -> dict[str, ArrayLike]:
```

Comment:
Oh, now the super class does exactly that. Remove?

Reply:
Done.